source/to_cpp1.h

```diff
@@ -1179,6 +1179,7 @@ class cppfront
         if (
             !sourcefile.ends_with(".cpp2")
             && !sourcefile.ends_with(".h2")
+            && sourcefile != "stdin"
```

Comment:
This part is fine, but we need to also update later logic that generates the output filename and currently assumes it ends in one of these and so just truncates the last character (e.g., `.cpp2` becomes `.cpp`. When I tried to use `stdin` the output went to a file named `stdi` (last character blindly truncated).

How is this intended to work -- if you use `stdin`, is the output expected to go to `stdout`, or if to a named file what is the intended name?

I think the right answer is probably to default to `stdout` unless the programmer overrode that with `-o`. I'll implement that, and also make sure the output to `stdout` is clean of other messages when there are no errors.